src/main/java/edu/hm/hafner/analysis/SecureXmlParserFactory.java

```diff
@@ -81,6 +74,30 @@ public DocumentBuilder createDocumentBuilder() {
         }
     }
 
+    private void setFeature(final DocumentBuilderFactory factory, final String enabledProperty, final boolean status) {
+        try {
+            factory.setFeature(enabledProperty, status);
+        }
+        catch (ParserConfigurationException ignored) {
+            // ignore and continue
+        }
+    }
+
+    private void secureFactory(final DocumentBuilderFactory factory) {
```

Comment:
Can you use the same mechanism here as well (inline with array and for loop)? It looks somehow strange that some values are set using an array and some are set in subsequent calls.

Reply:
Good idea.  I also created a method to set the features.